nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -89,6 +99,17 @@ public ValidationResult validate(final String subject, final String value, final
         .expressionLanguageSupported(true)
         .addValidator(DOCUMENT_VALIDATOR)
         .build();
+
+    static final AllowableValue LOC_BODY = new AllowableValue("body", "Body");
+    static final AllowableValue LOC_PARAM = new AllowableValue("param", "Query Parameter");
+    static final PropertyDescriptor QUERY_LOC = new PropertyDescriptor.Builder()
```

Comment:
Everything is tied to that parameter (or should be tied to it). One of the things I'm learning from having to train someone on NiFi is that "explicit is better" for a lot of people. I think the meaning of a blank query wouldn't be obvious to a lot of users, and you cannot trust that they'll read the description closely rather than cursing out the product because it seems to be "mysteriously misbehaving."

Reply:
We can ask the users list to get UX and other feedback, for ExecuteSQL it was natural because the query used to be required, then we relaxed it and added doc to the property saying if left blank, the contents of the flow file are expected to have the SQL to execute. In this case, a blank setting is already meaningful, so we may need two properties. "Explicit is better" is a good rule of thumb, but now we have two properties that interact with each other in different ways, which can also be confusing (see my other comments on when things are set to various value combos)

Reply:
Actually, I think allowing a blank field was the wrong decision because it is not a valid query w/ MongoDB. So I'll make that change.